store/store.go

```diff
@@ -1978,14 +1981,24 @@ func isValidAction(action string) bool {
 }
 
 type snapActionJSON struct {
-	Action           string `json:"action"`
-	InstanceKey      string `json:"instance-key"`
-	Name             string `json:"name,omitempty"`
-	SnapID           string `json:"snap-id,omitempty"`
-	Channel          string `json:"channel,omitempty"`
-	Revision         int    `json:"revision,omitempty"`
-	IgnoreValidation *bool  `json:"ignore-validation,omitempty"`
-}
+	Action           string      `json:"action"`
+	InstanceKey      string      `json:"instance-key"`
+	Name             string      `json:"name,omitempty"`
+	SnapID           string      `json:"snap-id,omitempty"`
+	Channel          string      `json:"channel,omitempty"`
+	Revision         int         `json:"revision,omitempty"`
+	Epoch            interface{} `json:"epoch,omitempty"` // see note
+	IgnoreValidation *bool       `json:"ignore-validation,omitempty"`
+}
+
+// NOTE about epoch in snapActionJSON: the store needs an epoch (even if null)
```

Comment:
I think this would be a better fit when placed right above `Epoch` field or above the whole struct.